### PR TITLE
Simplify testing and fix coverage

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

File: tests/test_outside_project_root.py [hunk 1]
```diff
@@ -10,20 +10,23 @@
...
-
-    exception_thrown = False
-    try:
-        should_fail()
-    except SKBuildError:
-        exception_thrown = True
-
-    assert exception_thrown
+
+    with push_dir():
```
Comment (opadron) [inline tests/test_outside_project_root.py]: This shouldn't be necessary, since the `project_setup_py_test` decorator should already take care of that.

Comment (jcfr) [inline tests/test_outside_project_root.py]: Without this, the `coverage.xml` file is generated in the test subdirectory.

**Comment by thewtex** [comment]
:+1:

**Comment by jcfr** [comment]
@thewtex @opadron   Thanks for review and integration

**Pull request merged**